pkg/fqdn/dnsproxy/proxy_test.go

```diff
@@ -800,17 +800,19 @@ func (s *DNSProxyTestSuite) TestFullPathDependence(c *C) {
 	jsn, err := json.Marshal(s.proxy.restored[epID1])
 	c.Assert(err, Equals, nil, Commentf("Could not marshal restored rules to json"))
 
+	// The fqdn regex matches backwards for performance reasons, so the regex
+	// is expected to look a bit strange
 	expected := `
 	{
 		"53": [{
-			"Re":  "(^[-a-zA-Z0-9_]*[.]ubuntu[.]com[.]$)|(^aws[.]amazon[.]com[.]$)",
+			"Re":  "^[.](?:moc[.]nozama[.]swa|moc[.]utnubu[.][-a-zA-Z0-9_]*)$",
 			"IPs": {"::": {}}
 		}, {
-			"Re":  "(^cilium[.]io[.]$)",
+			"Re":  "^[.](?:oi[.]muilic)$",
 			"IPs": {"127.0.0.1": {}, "127.0.0.2": {}}
 		}],
 		"54": [{
-			"Re":  "(^example[.]com[.]$)",
+			"Re":  "^[.](?:moc[.]elpmaxe)$",
 			"IPs": null
```

Comment:
One thing we can do if we are worried about the readability of the tests is to construct this string with `fmt.Sprintf`. I don't think this is too big of a deal, specifically as we have a comment explaining why it looks unexpected.